apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs

```diff
@@ -29,6 +29,17 @@
 
 namespace Google.Cloud.PubSub.V1;
 
+// The class follows following conventions for the field names:
+// - An "ackDeadline" is the amount of time before which the message must be acknowledged or have its lease extended.
+// - A suffix of "window" indicates the duration of time before the ackDeadline at which the lease is extended.
+// - A suffix of "interval" indicates the delay between sending one auto-lease extension request and the next; This is derived from the ackDeadline and the window.
+// The rules for manipulating the field values are:
+// - The "ackDeadline" should be within the MinAckDeadline and MaxAckDeadline. If not specified the DefaultAckDeadline is used.
```

Comment:
This is definitely better, but I wouldn't start talking about *which* min, max and default values are used. My intention was that this is effectively a set of rules for the *types* of values... exactly how those rules are used may well vary between EOD and non-EOD.